### ISSUE TITLE
Release 0.5.10 version of origin.js

### ISSUE THREAD

**Issue by wanderingstan** [OriginProtocol/origin-js#176]
This one's not on our usual cadence, but makes sense to do it early with all the awesome PR's that have come together, especially those unblocking other issues. 

## Confirm readiness
- [ ] _origin-js_: Confirm js tests passing
  - `git checkout develop`
  - `git pull`
  - `npm run install:dev`
  - `npm test` (This will also test smart contracts)
  - `npm start`
- [ ] _demo-dapp_: Confirm all tests passing
- [ ] _demo-dapp_: Confirm it runs against remote (unlinked) origin-js
  - `git checkout develop`
  - `git pull`
  - `npm install`
  - `npm start`
- [ ] _demo-dapp_: Confirm it runs against local (linked) origin-js
  - `git checkout develop`
  - `git pull`
  - `npm run install:dev`
  - `npm start`
  - Probably need to **Reset Account** on MetaMask. 
- [ ] Confirm deployment accounts have eth for gas
  - [ ] Rinkeby ([Faucet](https://faucet.rinkeby.io/)) 
    - Public post for [0]: `https://plus.google.com/u/0/114203467826302852415/posts/F7sJZMGCbgg`
    - [ ] account[0] `0x1a5c29c94d03c4c8f7414564cbd57295d61e898f`
    - [ ] account[1] `0x564aae0251d49d1f8d4d8d9e5da08f8cceff9ef2`
  - [ ] Ropsten ([Faucet](https://faucet.metamask.io/))
    - [ ] account[0] `0x1a5c29c94d03c4c8f7414564cbd57295d61e898f`
    - [ ] account[1] `0x564aae0251d49d1f8d4d8d9e5da08f8cceff9ef2`

## Publish
- [ ] _demo-dapp_: Replace git reference with version number in `package.json`
  -   ` "origin": "OriginProtocol/origin-js#develop"` --> `"origin": "0.6.0"`
- [ ] _origin-js_ : In `package.json`, update version to `0.6.0` 
- [ ] If contracts have changed:
  - Show diff with: `git diff master..develop contracts/contracts/`
  - `cd contracts`
  - [ ] Deploy new smart contracts to Ropsten. Be sure addresses are listed in ABI files. 
    - `export ROPSTEN_MNEMONIC="<mnemonic>"`
    - `npx truffle migrate --network ropsten`
  - [ ] Deploy new smart contracts to Rinkeby.  Be sure addresses are listed in ABI files. 
    - `export RINKEBY_MNEMONIC="<mnemonic>"`
    - `npx truffle migrate --network rinkeby`
  - [ ] Migrate data from old contracts to new. (Once we get around to writing migrations!)
- [ ] _origin-js_: Build origin.js (in `dist/origin.js`) -- this will bake in the new contract addresses.
    - `npm run install:dev`
- [ ] _demo-dapp_: Merge `develop` into `master` and push
- [ ] _origin-js_: Merge `develop` into `master` and push
- [ ] _origin-js_: Create new [GitHub release](https://github.com/OriginProtocol/origin-js/releases) with origin.js code,
  - [ ] Version in form `v0.6.0` (This will add git tag on `master`)
  - [ ] Include addresses of smart contracts in description
- [ ] _demo-dapp_: Add git tag to `master` to match origin-js.
  - `git tag -a v0.6.0 -m "New release"`
- [ ] _origin-js_: [Publish to npm](https://docs.npmjs.com/cli/publish). 
  - `npm publish`
- [ ] _demo-dapp_: Confirm that demo-dapp works when run alone again NPM. **FAILING**
- [ ] _demo-dapp_: Deploy new version of dapp to heroku (url to be determined)
- [ ] _bridge-server_: merge develop to master

## Follow-up
- [ ] Confirm published `origin.js` file is accessible via `code.originprotocol.com` redirect
  - https://unpkg.com/origin@0.6.0/dist/origin.js for now
- [ ] _demo-dapp_: Confirm that "one-line setup & run" command works on `master` branch shown by default
- [ ] _demo-dapp_: confirm its using npm package works and can find contracts
- [ ] _origin-js_: Increment version number to  on `develop` to for next release
- [ ] `git push`
- [ ] _demo-dapp_: Replace version number with git reference in `package.json`
  -`"origin": "0.6.0"` --> ` "origin": "OriginProtocol/origin-js#develop"`
  - `git push`
- [ ] _docs_: Review docs for needed updates. Confirm example code on playground site (jsfiddle?) still work.
- [ ] Copy this to-do list into new issue for next sprint.
- [ ] Post notice of new release on Discord

**Comment by wanderingstan** [comment]
***Ropsten***
```
$ npx truffle migrate --network ropsten
Using network 'ropsten'.

Running migration: 2_deploy_contracts.js
  Running step...
  Deploying PurchaseLibrary...
  ... 0xe10164584c5419dcec016c4e6980d71582c02dd8d97c4742fd02807abeda7505
  PurchaseLibrary: 0x2dcdbd31bae94b4311a7ee1a92a9f2ca6b90eac3
  Linking PurchaseLibrary to ListingsRegistry
  Linking PurchaseLibrary to Listing
  Deploying ListingsRegistry...
  ... 0x633e90be804944d46095b46bd1ed32440c6bbe079c4c6514b0d863b3d2d3c225
  ListingsRegistry: 0x2eeb6eafecef49a82d25a03d136c61109b61793a
  Deploying UserRegistry...
  ... 0xb943caea12021c97fc9cb45a228c6bdc9ac65303e69ad5d4b1a33eab7b5d4888
  UserRegistry: 0x71ca1ea454b7b7878d57a82dd767f06afc31e611
  Deploying OriginIdentity...
  ... 0xc704cb7c4d26d8fc1117561eaa19cd4cda33878091e14ea589e98199cc48ab37
  OriginIdentity: 0xabd7d77d3d635ea1040b82e1865de5b2bde2fe05
Saving successful migration to network...
  ... 0x60e1ca572be4c0d8f1b783d55dcfea96551c2a587871d777a9dbe585ba70d21c
Saving artifacts...
Running migration: 3_create_sample_listings.js
  Running step...
default_account:       0x1a5c29c94d03c4c8f7414564cbd57295d61e898f
a_seller_account:      0x564aae0251d49d1f8d4d8d9e5da08f8cceff9ef2
a_buyer_account:       0x47d4e5afd1624ea63cd7ace59d5dc1d87cff3bb5
another_buyer_account: 0xb7cc090637f6af85e3ed8cc623a5efe95b90b9f9
  ... 0x6ddea203688f71c74867f44369130093d3954d4f9b26112d01ea3ffc515bbb0a
  ... 0xb3b0d094e5f1ec9f7e3b57f34b34b92becc3d8cbc9f34763b12a5c65415ea89d
  ... 0xcdc2a9d74a92b2b822c2cc6243990fa57be4f4b342613b658582d5e3f45e5daf
  ... 0x4d8c4f5b7e2f346617e8c165848abb8b998a668e8a667aaa44c8b449e3c9cc40
  ... 0x08afe5538f9b0e9be0fca5d7289607c082d5a191cbbf1410242ccf50a506ed96
Saving successful migration to network...
  ... 0x77d8de454a91b29399cf971a1e317e509d3bb3b07fc8f18e79a5242be940f17f
Saving artifacts...
Running migration: 4_add_sample_issuers.js
  Running step...
  ... 0x8fc6310e5dc805fb3dc63960aa9423e9016a030346235cef444f6ae702b4d611
  ... 0x19bc5d9fec46798d3de93effbd09816b823870d887085dd260281088d2d53c1e
Saving successful migration to network...
  ... 0xfb94baa66292f286963a5cf7c92f60176766be9da8ef4d995617da771a0fc009
Saving artifacts...
```
